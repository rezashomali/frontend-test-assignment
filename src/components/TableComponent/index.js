const React = require("react");
import PropTypes from "prop-types";
import {
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Paper,
} from "@material-ui/core";

const TableComponent = ({ months, dataGood, dataMedian, dataBad }) => {
  const rows = months.map((entry, idx) => (
    <TableRow key={idx}>
      <TableCell>{entry}</TableCell>
      <TableCell>{dataGood[idx]}</TableCell>
      <TableCell>{dataMedian[idx]}</TableCell>
      <TableCell>{dataBad[idx]}</TableCell>
    </TableRow>
  ));

  return (
    <TableContainer component={Paper}>
      <table>
        <TableHead>
          <TableRow>
            <TableCell key="month">month</TableCell>
            <TableCell key="good">good</TableCell>
            <TableCell key="median">median</TableCell>
            <TableCell key="bad">bad</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>{rows}</TableBody>
      </table>
    </TableContainer>
  );
};

TableComponent.propTypes = {
  months: PropTypes.array,
  dataGood: PropTypes.array,
  dataMedian: PropTypes.array,
  dataBad: PropTypes.array,
};

export default TableComponent;
