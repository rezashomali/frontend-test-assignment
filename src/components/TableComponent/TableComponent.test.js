import React from "react";
import renderer from "react-test-renderer";
import TableComponent from "./index";

test("TableComponent component snapshot ", () => {
  const component = renderer.create(
    <TableComponent months={[]} dataGood={[]} dataMedian={[]} dataBad={[]} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
