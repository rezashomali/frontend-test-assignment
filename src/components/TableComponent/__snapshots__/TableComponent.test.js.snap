// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent component snapshot  1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table>
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          month
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          good
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          median
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          bad
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    />
  </table>
</div>
`;
